src/server/frontend_wayland/xdg_decoration_unstable_v1.cpp

```diff
@@ -127,7 +138,7 @@ void mir::frontend::XdgDecorationManagerV1::get_toplevel_decoration(wl_resource*
         BOOST_THROW_EXCEPTION(std::runtime_error("Invalid toplevel pointer"));
     }
 
-    auto decoration = new XdgToplevelDecorationV1{id, tl};
+    auto decoration = new XdgToplevelDecorationV1{id, tl, std::move(decoration_strategy)};
```

Comment:
I had this exact same thought, but I tried confirming my doubts by running the code with multiple windows and nothing crashed or misbehaved. I think I should've asked before accidentally introducing a subtle bug into our code. 

I've got two questions:
1. Why didn't the code crash? Could it be that my crude testing method was incorrect, or perhaps something else?
2. I'm not sure I understand what  XdgDecorationManagerV1::decoration_strategy being const has to do with calling `std::move`? Is it because moving modifies the shared pointer, which shouldn't happen with a const object?

Reply:
So I just ran the code under GDB. The use count increases normally and the stored pointer is not nulled. Digging a bit further, it's seems that the copy constructor is being called under the hood...

Reply:
Sorry, I wasn't clear. The behaviour of `std::move()` on a `const` source is surprising (to me at least): It returns a `const ...&&`. So the copy constructor is called as you observe.

But the reader of the code may well be misled into thinking that move semantics change the source

Reply:
Indeed, removing the `const` causes the move constructor to be called and Mir to crash. Sorry! Should be fixed in the latest commit.